### ISSUE TITLE
ASHA Promotion Fails

### ISSUE THREAD

**Issue by wistuba** [awslabs/syne-tune#395]
If you run the following example, this error is raise: `AssertionError: Internal error (trial_id 2): on_trial_result called with resource = 1, but largest_update_resource = 3`
`type=stopping` works.

```
from pathlib import Path

from syne_tune.backend import LocalBackend
from syne_tune.optimizer.baselines import ASHA
from syne_tune import Tuner, StoppingCriterion
from syne_tune.config_space import randint

config_space = {
    "steps": 100,
    "width": randint(0, 20),
    "height": randint(-100, 100),
}
entry_point = (
    Path("examples")
    / "training_scripts"
    / "height_example"
    / "train_height.py"
)

scheduler = ASHA(
    config_space,
    metric="mean_loss",
    resource_attr="epoch",
    max_resource_attr="steps",
    type="promotion",
    mode="min",
)

tuner = Tuner(
    trial_backend=LocalBackend(entry_point=str(entry_point)),
    scheduler=scheduler,
    stop_criterion=StoppingCriterion(max_wallclock_time=20),
    n_workers=4,
)

tuner.run()
```

**Comment by mseeger** [comment]
This should become a unit test once this is fixed, preferably with smaller max_wallclock_time.

We already have tst/test_schedulers.py, but apparently the time there is too short.

**Comment by mseeger** [comment]
Fixed by #399
